### PR TITLE
packages/phist: new version 1.7.4 with new variant 'host' to enable/d…

### PR DESCRIPTION
**Pull request by jthies**
…isable host-specific optimizations.

### CODE REVIEW

File: var/spack/repos/builtin/packages/phist/package.py [hunk 1]
```diff
@@ -72,6 +77,12 @@ class Phist(CMakePackage):
             description='generate Fortran 2003 bindings (requires Python3 and '
                         'a Fortran compiler)')
 
+    # in older versions, it is not possible to completely turn off OpenMP
+    conflicts('~openmp', when='@:1.7.3')
+    # in older versions, it is not possible to turn off the use of host-
+    # specific compiler flags in Release mode.
+    conflicts('~host', when='@:1.7.4')
+
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/phist/package.py]: flake complaint here (excess whitespace)

File: var/spack/repos/builtin/packages/phist/package.py [hunk 2]
```diff
@@ -127,6 +138,8 @@ def cmake_args(self):
                 % ('ON' if '+trilinos' in spec else 'OFF'),
                 '-DXSDK_ENABLE_Fortran:BOOL=%s'
                 % ('ON' if '+fortran' in spec else 'OFF'),
+                '-DPHIST_HOST_OPTIMIZE:BOOL=%s'
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/phist/package.py]: Given the conflict above, does this keyword exist before version 1.7.4?

**Comment by balay** [comment]
I've attempted a build with`~host`. I see `'-DPHIST_HOST_OPTIMIZE:BOOL=OFF'` passed to cmake. But then - the build log has compile commands with: `-march=native`.

Attaching log
[build.out.txt](https://github.com/spack/spack/files/2626886/build.out.txt)

**Comment by jthies** [comment]
the ~host option should now work better, I fixed an issue in phist and moved the phist-1.7.4 tag to the llatest devel head

**Comment by balay** [comment]
Thanks! I don't see `-march` in build log now.

BTW: Could you also fix `conflicts('~host', when='@:1.7.3')`?

**Pull request closed without merging**